### ISSUE TITLE
Regression: mac test failure

### ISSUE THREAD

**Issue by beldmit** [openssl/openssl#12837]
Using gost engine:

./openssl_wrap.sh dgst -md_gost12_512 -hmac 123456901234567890123456789012 somefile

the exit code is 1 instead of 0, nothing is written to stderr, no output.

The openssl_wrap.sh script:
```
#!/bin/bash
  
OPENSSLDIR=`pwd`/../openssl
export LD_LIBRARY_PATH=$OPENSSLDIR
OPENSSL_CONF=`pwd`/engine.conf $GDB $OPENSSLDIR/apps/openssl $@
```

engine.conf file:

```
openssl_conf = openssl_def
[openssl_def]
engines = engine_section

[engine_section]
gost = gost_section

[gost_section]
engine_id = gost
dynamic_path = /path/to/gost.so
default_algorithms = ALL
CRYPT_PARAMS = id-Gost28147-89-CryptoPro-A-ParamSet

[req]
distinguished_name = req_dn
[ req_dn ]
C = RU
L = Moscow
CN=Test srv gost2012_256:XA
O=Cryptocom
OU=OpenSSL CA
emailAddress = openssl@cryptocom.ru
```

**Comment by beldmit** [comment]
The bug was introduced recently, #12740 fixed all the issues present at that moment.

**Comment by mattcaswell** [comment]
Fix in #12850.
